### ISSUE TITLE
Dependency Conflict

### ISSUE THREAD

**Issue by ghost** [KevinMcK100/binance-dynamic-savings-bot#1]
ERROR: Cannot install -r requirements.txt (line 14) and regex==2022.3.15 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested regex==2022.3.15
    dateparser 1.1.1 depends on regex!=2019.02.19, !=2021.8.27 and <2022.3.15

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

**Note: I updated packages requirements.txt with pur prior to this, but this still happens when using the provided requirements.txt, see below for an error stemming from using the original req file.**

ERROR: Cannot install python-binance==1.0.15 and websockets==10.2 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested websockets==10.2
    python-binance 1.0.15 depends on websockets==9.1

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

**Comment by KevinMcK100** [comment]
Hi. Yes I'm aware of the dependency issues. I haven't yet experienced the regex issue but did find the websockets issue. Problem is each library requires different versions of websockets. 

I was able to work around this temporarily by using `pip install <PACKAGE>` for all required dependencies within my virtual env. I think all you need is numpy, pandas, python-binance, python-telegram-bot, PyYAML, scheduler, unicorn-binance-websocket-api. With this manual method it installs websockets 10.2 which allows it to work fine for unicorn-binance-websocket-api but I'm sure it would fail if I tried using websockets in python-binance with this version.

2 things I'm planning to do to resolve these dependency issues:

1. Given that python-binance has support for websockets, I will attempt to just use it for both API calls and order streaming and then remove unicorn-binance-websocket-api from the project completely. This solves the websockets dependency issue.
2. I also plan to try using `pip-tools` with `pip-compile` rather than `pip freeze > requirements.txt`. I believe this should do a better job of figuring out the required transient dependencies (hopefully solving the regex issue pointed out above)

One other note. This is project is currently in development/testing. If you install and run it now it will not perform any real operations on your Spot/Flexible Savings account (other than fetch and monitor real order and savings data from your account). However it is relatively close so should be fully functional in a few days.

**Comment by KevinMcK100** [comment]
Dependency issues should now be resolved. I have removed the use of `unicorn-binance-websocket-api` completely and am only using `python-binance` now.

I've also updated the start script to use `pip-compile` which uses the new `requirements.in` file to generate new `requirements.txt`.

Also I found that Python version >=3.9 is now required due to using the `scheduler` library which requires this: https://github.com/DigonIO/scheduler/blob/master/setup.py#L46
